spring-jdbc/src/main/java/org/springframework/jdbc/core/metadata/TableMetaDataContext.java

```diff
@@ -265,18 +266,19 @@ public List<Object> matchInParameterValuesWithInsertColumns(SqlParameterSource p
 	 * @param inParameters the parameter names and values
 	 */
 	public List<Object> matchInParameterValuesWithInsertColumns(Map<String, ?> inParameters) {
-		List<Object> values = new ArrayList<>(inParameters.size());
+		List<Object> values = new ArrayList<>(this.tableColumns.size());
+		
+		Map<String, Object> caseInsensitiveLookup = new HashMap<>(inParameters.size());
+		for (Map.Entry<String, ?> entry : inParameters.entrySet()) {
+			caseInsensitiveLookup.put(entry.getKey().toLowerCase(Locale.ROOT), entry.getValue());
+		}
+		
 		for (String column : this.tableColumns) {
 			Object value = inParameters.get(column);
 			if (value == null) {
 				value = inParameters.get(column.toLowerCase(Locale.ROOT));
 				if (value == null) {
-					for (Map.Entry<String, ?> entry : inParameters.entrySet()) {
-						if (column.equalsIgnoreCase(entry.getKey())) {
-							value = entry.getValue();
-							break;
-						}
-					}
+					value = caseInsensitiveLookup.get(column.toLowerCase(Locale.ROOT));
```

Comment:
When using the `LinkedCaseInsensitiveMap` you can just do a regular `get` without the need to convert to lowercase (the map will take care of this). Which would simplify this code.